### ISSUE TITLE
Show ACM fails when auth rules are distinguished only by their groupClaim

### ISSUE THREAD

**Issue by naedx** [aws-amplify/amplify-category-api#537]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.18.2

### Amplify CLI Version

8.4.0

### What operating system are you using?

Windows

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

auth, api

### Amplify Commands

status

### Describe the bug

The command `amplify status api -acm Type` fails for schema where the two or more auth rules are defined but are distinguished only by the name of the `groupClaim`.

```
PS *project-path*> amplify status api -acm Blog 
✅ GraphQL schema compiled successfully.

Edit your schema at *project-path*\amplify\backend\api\testproject1\schema.graphql or place .graphql files in a directory at *project-path*\amplify\backend\api\testproject1\schema
🛑 @auth userPools:dynamicGroup:tenantId already exists for Blog

```

### Expected behavior

The command should print the ACM without exception.

### Reproduction steps

1. Create a new amplify project 
2. Add category auth with default options
3. Add category api with cognito auth and the schema below.
4. Execute `amplify status api -acm Blog`

### GraphQL schema(s)

<details>

```graphql

type Blog 
  @model 
  @auth(rules: [
    { allow: groups, groupsField: "tenantId", groupClaim: "custom:adminRole" }
    { allow: groups, groupsField: "tenantId", groupClaim: "custom:editorRole", operations: [read, update] }
  ])
{
  id: ID!
  name: String!
  tenantId: String!
}


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by josefaidt** [comment]
Hey @naedx :wave: thanks for raising this! I was able to successfully reproduce this issue using the provided steps, thank you for including those details! Marking as a bug 🙂

**Comment by naedx** [comment]
Hey @josefaidt . Thanks for verifying the bug. 

I have idenified the cause and have [proposed a fix with PR #10577](https://github.com/aws-amplify/amplify-cli/pull/10577).

**Comment by sundersc** [comment]
Closing due to inactivity, feel free to reopen if you run into the issue.
